### DISCUSSION TITLE
Is there a functional difference using __init__()  and parse_obj() for creating a resource from a dictionary

### DISCUSSION THREAD

**Discussion by df65** [nazrulworld/fhir.resources#113] [Q&A] [answered]
First of all thanks @nazrulworld for this library.  A newbie question: 

Can you advise if there is a difference using __init__()  and parse_obj() for creating a resource (apart from the latter using a class method to create the object)?

In the code below,  p1 & p2 seem identical.  Does one have different validation and/or performance than the other?

    p_obj = { "resourceType":"Patient", "id":"p1", "active":True, "name":[{"text": "Adam Smith"}], "birthDate": "1985-06-12"}
    p1 = Patient.parse_obj(p_obj)
    p2 = Patient(**p_obj)

**Comment by axelv** [comment, marked as answer]
Hi @df65,

`fhir.resources` is built on top of [`pydantic`](https://docs.pydantic.dev/) and I think your question is mainly about how `pydantic` models work.

This page probably answers your question: [https://docs.pydantic.dev/usage/models/#helper-functions](https://docs.pydantic.dev/usage/models/#helper-functions)
> parse_obj: this is very similar to the __init__ method of the model, except it takes a dict rather than keyword arguments. If the object passed is not a dict a ValidationError will be raised.



>

**Reply by df65** [reply]
Thank you Axel.
